conans/client/graph/proxy.py

```diff
@@ -33,10 +33,14 @@ def _get_recipe(self, reference, remotes, update, check_update):
             return conanfile_path, RECIPE_EDITABLE, None, reference
 
         # check if it there's any revision of this recipe in the local cache
-        ref = self._cache.get_latest_recipe_reference(reference)
+        try:
```

Comment:
Not necessary to do 2 DB calls, first get the reference, then the necessary layout, but we can get in one db call the whole layout.